src/platforms/common/server/owning_c_str.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdlib>
+#include <memory>
+
+namespace
+{
+class CStringFree
+{
+public:
+    void operator()(char* str)
+    {
+        if(str)
+        {
+            free(str);
+        }
+    }
+};
+}
+
+namespace mir
+{
+using CStr = std::unique_ptr<char[], CStringFree>;
+}
```

Comment:
1. what? No header guards?
2. The null check is redundant
3. The `operator()` can be `const`
4. This doesn't handle `const` c_strings

```suggestion
#ifndef MIR_OWNING_C_STR_
#define MIR_OWNING_C_STR_

#include <cstdlib>
#include <memory>

namespace mir
{
using CStr = std::unique_ptr<char[], decltype([](char const* p) { ::free(const_cast<char *>(p)); })>;
}

#endif
```